src/Symfony/Component/Translation/DataCollectorTranslator.php

```diff
@@ -88,19 +88,30 @@ public function getCatalogue($locale = null)
     }
 
     /**
-     * Gets the fallback locales.
-     *
-     * @return array $locales The fallback locales
+     * {@inheritdoc}
      */
     public function getFallbackLocales()
     {
-        if ($this->translator instanceof Translator || method_exists($this->translator, 'getFallbackLocales')) {
+        if ($this->translator instanceof TranslatorFallbackInterface) {
```

Comment:
We would still need to check for the method. Otherwise we would drop support for translator implementations which provide the method, but do not implement the new interface.

Reply:
It's probably a good idea to trigger a deprecation though, if the wrapped translator implements the method but not the interface